### PR TITLE
Ah/moleculargraph support

### PR DESCRIPTION
**Pull request by anhi**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ DataFramesMeta = "1313f7d8-7da2-5740-9ea0-a2ca25f37964"
 EnumX = "4e289a0a-7415-4d19-859d-a7e5c4648b56"
 JSON3 = "0f8b85d8-7281-11e9-16c2-39a750bddbf1"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+MolecularGraph = "6c89ec66-9cd8-5372-9f91-fabc50dd27fd"
```
Comment (tkemmer) [inline Project.toml:15]: The corresponding compat entry is missing.

Comment (anhi) [reply Project.toml:15]: fixed.

File: src/substructures/smarts.jl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        new(query, smartstomol(query))
+    end
+end
+
+function _to_substructure(mol, m; adjacent_bonds=false)
+    matched_atoms = keys(m)
+
+    atom_view = filter(:number => n -> n ∈ matched_atoms, mol.atoms, view=true)
+    bond_view = filter(
+        [:a1, a2] => (a1, a2) ->
```
Comment (tkemmer) [inline src/substructures/smarts.jl:20]: `ERROR: UndefVarError: a2 not defined`

File: src/substructures/smarts.jl [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    matched_atoms = keys(m)
+
+    atom_view = filter(:number => n -> n ∈ matched_atoms, mol.atoms, view=true)
+    bond_view = filter(
+        [:a1, a2] => (a1, a2) -> 
+            adjacent_bonds ? a1 ∈ matched_atoms && a2 ∈ matched_atoms
+                           : a1 ∈ matched_atoms || a2 ∈ matched_atoms,
+        mol.bonds, view=true)
+
+    Substructure("$(query) on $(mol.name)", atom_view, bond_view)
```
Comment (tkemmer) [inline src/substructures/smarts.jl:25]: `ERROR: UndefVarError: query not defined`

Comment (tkemmer) [reply src/substructures/smarts.jl:25]: Also, the constructor call requires additional arguments (parent molecule and properties).

Comment (anhi) [reply src/substructures/smarts.jl]: good catch. that broke during the rebase. should be fixed now

**Pull request closed without merging**